run_kraken.py

```diff
@@ -1,15 +1,111 @@
+    instance_kill_count = node_scenario.get("instance_kill_count", 1)
+    node_name = node_scenario.get("node_name", "")
+    label_selector = node_scenario.get("label_selector", "")
+    timeout = node_scenario.get("timeout", 120)
+
+    # Get the node to apply the scenario
+    node = nodeaction.get_node(node_name, label_selector)
+
+    if action == "node_start_scenario":
+        node_scenario_object.node_start_scenario(instance_kill_count, node, timeout)
+    elif action == "node_stop_scenario":
+        node_scenario_object.node_stop_scenario(instance_kill_count, node, timeout)
+    elif action == "node_stop_start_scenario":
+        node_scenario_object.node_stop_start_scenario(instance_kill_count, node, timeout)
+    elif action == "node_termination_scenario":
+        node_scenario_object.node_termination_scenario(instance_kill_count, node, timeout)
+    elif action == "node_reboot_scenario":
+        node_scenario_object.node_reboot_scenario(instance_kill_count, node, timeout)
+    elif action == "stop_kubelet_scenario":
+        node_scenario_object.stop_kubelet_scenario(instance_kill_count, node, timeout)
+    elif action == "stop_start_kubelet_scenario":
+        node_scenario_object.stop_start_kubelet_scenario(instance_kill_count, node, timeout)
+    elif action == "node_crash_scenario":
+        node_scenario_object.node_crash_scenario(instance_kill_count, node, timeout)
+
+
+# Inject the cluster shut down scenario
+def cluster_shut_down(shut_down_config, cerberus_enabled, config):
+    runs = shut_down_config["runs"]
+    shut_down_duration = shut_down_config["shut_down_duration"]
+    cloud_type = shut_down_config["cloud_type"]
+    if cloud_type == "aws":
+        cloud_object = AWS()
+
+    nodes = set(kubecli.list_nodes())
+    node_id = {}
+    for node in nodes:
+        node_id[node] = cloud_object.get_instance_id(node)
+
+    for _ in range(runs):
+        logging.info("Starting cluster_shut_down scenario injection")
+        for node in nodes:
+            cloud_object.stop_instances(node_id[node])
+        logging.info("Waiting for 250s to shut down all the nodes")
+        time.sleep(250)
+        logging.info("Shutting down the cluster for the specified duration: %s"
+                     % (shut_down_duration))
+        time.sleep(shut_down_duration)
+        logging.info("Restarting the nodes")
+        restarted_nodes = set()
+        stopped_nodes = nodes
+        while restarted_nodes != nodes:
+            for node in stopped_nodes:
+                try:
+                    cloud_object.start_instances(node_id[node])
+                    restarted_nodes.add(node)
+                except Exception:
+                    time.sleep(10)
+                    continue
+            stopped_nodes = nodes - restarted_nodes
+        logging.info("Waiting for 250s to allow cluster component initilization")
+        time.sleep(250)
+        logging.info("Successfully injected cluster_shut_down scenario!")
```

Comment:
Are we able to add in a verification that the nodes are all back up and ready here? Is that too much for kraken that it can just be handled in cerberus? Thoughts?

Reply:
I think this part would be handled by cerberus. With cerberus intergration, when we receive a true, kraken proceeds with next scenario indicating all the nodes are ready but with false, we terminate kraken indicating some components aren't healthy. But it can be explicitly specified after [line](https://github.com/openshift-scale/kraken/pull/25/files#diff-cfec6e969439720ae75ffa7156ac298bR107) when cerberus integration is enabled if needed. Thoughts?